clients/include/testing_trmm.hpp

```diff
@@ -27,138 +28,177 @@ void testing_trmm(const Arguments& arg)
-        static const size_t safe_size = 100;
+        static const size_t safe_size = 100; // arbitrarily set to 100
         device_vector<T>    dA(safe_size);
         device_vector<T>    dB(safe_size);
-        device_vector<T>    dC(safe_size);
-        if(!dA || !dB || !dC)
+        if(!dA || !dB)
         {
             CHECK_HIP_ERROR(hipErrorOutOfMemory);
             return;
         }
 
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
         EXPECT_ROCBLAS_STATUS(
-            rocblas_trmm<T>(
-                handle, side, uplo, transA, diag, M, N, &alpha, dA, lda, dB, ldb, dC, ldc),
+            rocblas_trmm<T>(handle, side, uplo, transA, diag, M, N, &alpha_h, dA, lda, dB, ldb),
             rocblas_status_invalid_size);
-
         return;
     }
+
     // Naming: dK is in GPU (device) memory. hK is in CPU (host) memory
     host_vector<T> hA(size_A);
     host_vector<T> hB(size_B);
-    host_vector<T> hC(size_B);
-    host_vector<T> hB_copy(size_B);
+    host_vector<T> hB_1(size_B);
+    host_vector<T> hB_2(size_B);
+    host_vector<T> cpuB(size_B);
 
     double gpu_time_used, cpu_time_used;
     double rocblas_gflops, cblas_gflops;
-    double rocblas_error;
-
-    rocblas_local_handle handle;
+    double rocblas_error = 0.0;
 
     // allocate memory on device
-    // dB and dC are exact the same size
     device_vector<T> dA(size_A);
     device_vector<T> dB(size_B);
-    device_vector<T> dC(size_B);
-    if(!dA || !dB || !dC)
+    device_vector<T> alpha_d(1);
+    if(!dA || !dB || !alpha_d)
     {
         CHECK_HIP_ERROR(hipErrorOutOfMemory);
         return;
     }
 
-    // Initial Data on CPU
-    rocblas_seedrand();
-    rocblas_init_symmetric<T>(hA, K, lda);
+    //  initialize full random matrix hA with all entries in [1, 10]
+    rocblas_init<T>(hA, K, K, lda);
+
+    //  pad untouched area into zero
+    for(int i = K; i < lda; i++)
+        for(int j = 0; j < K; j++)
+            hA[i + j * lda] = 0.0;
+
+    // Initial hX
     rocblas_init<T>(hB, M, N, ldb);
-    hB_copy = hB;
+    // pad untouched area into zero
+    for(int i = M; i < ldb; i++)
+        for(int j = 0; j < N; j++)
+            hB[i + j * ldb] = 0.0;
+
+    hB_1 = hB; // hXorB <- B
+    hB_2 = hB; // hXorB <- B
+    cpuB = hB; // cpuB <- B
 
     // copy data from CPU to device
     CHECK_HIP_ERROR(hipMemcpy(dA, hA, sizeof(T) * size_A, hipMemcpyHostToDevice));
-    CHECK_HIP_ERROR(hipMemcpy(dB, hB, sizeof(T) * size_B, hipMemcpyHostToDevice));
 
-    /* =====================================================================
-           ROCBLAS
-    =================================================================== */
-    if(arg.timing)
+    if(arg.unit_check || arg.norm_check)
     {
-        gpu_time_used = get_time_us(); // in microseconds
-    }
+        // calculate dB <- A^(-1) B   rocblas_device_pointer_host
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
```

Comment:
Looks like we're only doing host_mode tests and not any device_mode tests, is that intended?

Reply:
Thanks for catching this, added back device_mode tests.